.github/workflows/workflow-cleanup.yml

```diff
@@ -0,0 +1,47 @@
+## Github workflow to clean up old caches and workflow runs
+
+name: Workflow Cleanup
+
+on:
+  workflow_dispatch:
+    inputs:
+      cache-ttl:
+        description: "How many days to keep a cache (default: 7)"
+        required: false
+        default: "7"
+      workflow-ttl:
+        description: "How many days to keep a successful workflow (default: 30)"
+        required: false
+        default: "30"
+      failed-workflow-ttl:
+        description: "How many days to keep failed workflows (default: 15)"
+        required: false
+        default: "15"
+  schedule:
+    ## Run every day at 00:00:00
+    - cron: "0 0 * * *"
```

Comment:
IIRC, the cron respects UTC. So this will run at 7PM ET and 4PM PT. Should this be pushed back a bit to account for that considering most core engineers are around PT/ET?

Reply:
it is UTC, correct. 
It should not matter *when* it runs though, the outcome is still the same and i feel the defaults defined in https://github.com/stacks-network/actions/blob/main/cleanup/workflows/action.yml#L12-L23  are generous enough to not interfere with any current work. 

I expect this workflow will have some of the strongest opinions, so i'm happy to adjust/disable, or even change it to a manual trigger only. 

to be clear, this workflow only runs to clean up old actions/failed actions/caches:
- caches (excepting bitcoin binary are kept for 7 days ) *in the previous PR, you noted these will be rotated as space is needed as well. i saw no reason to remove this step, but it's debatable if it's needed*
- failed workflows are kept for 15 days
- successful workflows are kept for 30 days